### ISSUE TITLE
How could I add extra pre-processing operations only to the input of Discriminator?

### ISSUE THREAD

**Issue by HJLYU0519** [junyanz/pytorch-CycleGAN-and-pix2pix#505]
Sorry for asking this question again in this code.  I just realized that I used this code to achieve my ideas.

I made some changes to the Discriminator. And I want to add some extra pre-processing operations only to the input of Discriminator. What should I change?
And I think there are no preprocessing operations. my reason is this: in data/init.py
75 self.dataloader = torch.utils.data.DataLoader(
76 self.dataset,
77 batch_size=opt.batch_size,
78 shuffle=not opt.serial_batches,
79 num_workers=int(opt.num_threads))

Should I establish another dataloader only for Discriminator? Or  adding some operation before Discreiminator?
Many thanks for your kind help

**Comment by HJLYU0519** [comment]
I think that the generatored image (fake_A，fake_B) also need to be preprocessed, and then be added to the original dataset before sent to Discriminator. Because the original dataset is not incomplete (the lack of generated data from generator). I really want to know which line of code implements this above operation.

If I have any mistake, please point out directly. Thanks so much and your suggestion would be helpful for me.

**Comment by emckenzi123** [comment]
I am also interested in this problem

**Comment by junyanz** [comment]
You don't need to modify the dataloader. You can just add a pre-processing function in the training code, for example, this [line](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/models/pix2pix_model.py#L94).
